Reading List/Reading List/ReadingListTableViewController.swift

```diff
@@ -0,0 +1,113 @@
+    }
+    
+    func toggleHasBeenRead(for cell: BookTableViewCell) {
+        
+        if let book = cell.book {
+            bookController.updateRead(with: book)
+            tableView.reloadData()
+        }
+    }
+    
+    // MARK: - Table view data source
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return 2
+    }
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        // #warning Incomplete implementation, return the number of rows
+        if section == 0 {
+            return bookController.readBooks.count
+        } else {
+            return bookController.unreadBooks.count
+        }
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: "BookCell", for: indexPath) as? BookTableViewCell else {
+            return UITableViewCell()}
+        
+        let book = bookFor(indexPath: indexPath)
+        cell.book = book
+        cell.delegate = self
+        
+        return cell
+    }
+    
+    override func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
+        if editingStyle == .delete {
+            let book = bookFor(indexPath: indexPath)
+            bookController.deleteBook(with: book)
+            tableView.deleteRows(at: [indexPath], with: .fade)
+        }
+    }
+    
+    override func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        if section == 0 {
+            if bookController.readBooks.count > 0 {
+                return "Read Books"
+            } else {
+                return nil
+            }
+            
+        } else {
+            if bookController.unreadBooks.count > 0 {
+                return "Unread Books"
+            } else {
+                return nil
+            }
+        }
+    }
+    
+    // MARK: - Navigation
+    // In a storyboard-based application, you will often want to do a little preparation before navigation
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if segue.identifier == "AddBookSegue" {
+            guard let bookDetailVC = segue.destination as? BookDetailViewController else {return}
+            bookDetailVC.bookController = bookController
+            bookDetailVC.delegate = self
+        } else {
+            if segue.identifier == "BookDetailSegue" {
+                guard let bookDetailVC = segue.destination as? BookDetailViewController else {return}
+                bookDetailVC.bookController = bookController
+                guard let newIndex = self.tableView.indexPathForSelectedRow else {return}
+                let booked = bookFor(indexPath: newIndex)
+                bookDetailVC.book = booked
+                bookDetailVC.delegate = self
+            }
+        }
+    }
```

Comment:
Nice job not letting this get over indented. One thing, it is convention to put space inside of your return braces like this: `{ return }`